server/routes/review/visitsReviewListingController.ts

```diff
@@ -12,18 +14,49 @@ export default class VisitsReviewListingController {
   public view(): RequestHandler {
     return async (req, res) => {
       const { prisonName } = req.session.selectedEstablishment
-      const visitsReviewList = await this.visitNotificationsService.getVisitsReviewList(
+      // OK to cast these as express-validator sanitises to string[]
+      const bookedBy = Array.isArray(req.query.bookedBy) ? <string[]>req.query.bookedBy : []
+      const type = Array.isArray(req.query.type) ? <string[]>req.query.type : []
+
+      const { filters, visitsReviewList } = await this.visitNotificationsService.getVisitsReviewList(
         res.locals.user.username,
         req.session.selectedEstablishment.prisonId,
+        { bookedBy, type },
       )
 
+      const isAFilterApplied = filters.some(filter => filter.items.some(item => item.checked))
+
       return res.render('pages/review/visitsReviewListing', {
         notificationTypes,
         notificationTypeDescriptions: Object.values(notificationTypeDescriptions),
         notificationTypePathSegments,
         prisonName,
+        filters,
         visitsReviewList,
+        isAFilterApplied,
       })
     }
   }
+
+  public submit(): RequestHandler {
+    return async (req, res) => {
+      // OK to cast these as express-validator sanitises to string[]
+      const bookedBy = Array.isArray(req.body.bookedBy) ? <string[]>req.body.bookedBy : []
+      const type = Array.isArray(req.body.type) ? <string[]>req.body.type : []
+
+      return res.redirect(
+        url.format({
+          pathname: '/review',
+          query: {
+            ...(bookedBy && { bookedBy }),
+            ...(type && { type }),
+          },
+        }),
+      )
+    }
+  }
+
+  public validate(): ValidationChain[] {
+    return [check('bookedBy').toArray(), check('type').toArray()]
```

Comment:
Only sanitising here; haven't bothered with any _validation_ because list of allowed values depends on the data and invalid values are just discarded when data is filtered; or it'll just show data with no filters applied.